nixos/modules/services/hardware/nvidia-container-toolkit-cdi-generator/cdi-generate.nix

```diff
@@ -0,0 +1,39 @@
+{ config, lib, pkgs }: let
+  mountOptions = { options = ["ro" "nosuid" "nodev" "bind"]; };
+  mounts = [
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-control";
+      containerPath = "/usr/bin/nvidia-cuda-mps-control"; }
+    { hostPath = "${lib.getBin config.hardware.nvidia.package}/bin/nvidia-cuda-mps-server";
```

Comment:
We don't have means of extending this list locally. Can a user add a static config with more mounts that docker/podman would "merge" with the dynamic configuration?

Reply:
No, I didn't want to overengineer the solution. The static config would not merge in any case. According to the CDI spec, if anything happens, dynamic configuration takes precedence over static configuration. How merges specifically happen, or if they happen at all, I am not sure.

I think as a first approach having dynamic "works out of the box" and static gives a very good amount of flexibility to the user. As you have mentioned, maybe providing some settings to disallow mounting `/usr/bin` on the container is fine, but getting all the way to merging strategies and such, I don't think there's need for that, at least for now, until we start to gather feedback from more users.

Reply:
Overriding and extending the mount paths was the first my very first need: we expected we'd fail to account for all possible use-cases, and we did - we failed to account for `addDriverRunpath`. This one is trivial, but I don't have any doubts we'll encounter more.

IMO, ensuring the user's ability to repair things without patching nixpkgs should be the first priority

Reply:
> This one is trivial, but I don't have any doubts we'll encounter more.

I'd prefer to wait for that feedback. There's a way for users to workaround the issue by using static CDI config.

In any case, if you are convinced that we need this right now, I don't oppose, although I don't agree.